lib/Service/NotesService.php

```diff
@@ -30,11 +30,14 @@ public function __construct(
 		$this->noteUtil = $noteUtil;
 	}
 
-	public function getAll(string $userId, bool $autoCreateNotesFolder = false) : array {
+	public function getAll(string $userId, bool $autoCreateNotesFolder = false, ?bool $showHidden = null) : array {
 		$customExtension = $this->getCustomExtension($userId);
 		try {
 			$notesFolder = $this->getNotesFolder($userId, $autoCreateNotesFolder);
-			$data = self::gatherNoteFiles($customExtension, $notesFolder);
+			if ($showHidden === null) {
+				$showHidden = $this->settings->get($userId, 'showHidden');
```

Comment:
We now incur in an extra `SettingsService->get()` call (here and in `countNotes()`), but I suspect/expect (without deep knowledge of NC's internals) that settings retrieval might be backed by some cache?